src/NServiceBus.RavenDB.Tests/SubscriptionStorage/When_receiving_duplicate_subscription_messages.cs

```diff
@@ -48,14 +68,41 @@ public class When_receiving_duplicate_subscription_messages : RavenDBPersistence
         await storage.Subscribe(subscriber_v6_2, messageType, new ContextBag());
 
         WaitForIndexing(store);
- 
+
         var subscriber = (await storage.GetSubscriberAddressesForMessage(new[]
         {
             messageType
         }, new ContextBag())).ToArray();
- 
+
         Assert.AreEqual(1, subscriber.Length);
         Assert.AreEqual(subscriberAddress, subscriber[0].TransportAddress);
         Assert.AreEqual("new_endpoint_name", subscriber[0].Endpoint);
     }
+
+    [Test]
+    public async Task should_overwrite_existing_subscription_event_if_event_version_changed()
```

Comment:
not sure if we discussed this in the end, but since version number is part of document id, in theory you can have multiple documents for one message type, then which one should be updated?
shouldn't we create a new one anyway that doesn't include major version in the generated id (which is what happens now)?